### ISSUE TITLE
Is it possible to add icons in the first column and bind events to them?

### ISSUE THREAD

**Issue by 851091009** [quicktype/glide-data-grid#767]
![image](https://github.com/quicktype/glide-data-grid/assets/8460036/f5464f70-6c52-4160-b6c3-32d86431ccc0)

**Comment by jassmith** [comment]
you want these in the row marker?

**Comment by 851091009** [comment]
> you want these in the row marker?

Yes, it would be great if it could be implemented. This way, other operations can be performed. For example, clicking on the icon can trigger a pop-up editing box.

![table-icon](https://github.com/quicktype/glide-data-grid/assets/8460036/df18aae5-1220-4af2-9c0a-8d9c8a965885)

**Comment by jassmith** [comment]
I see... uh... I mean... I wouldn't hold your breath. I definitely love this concept but I dont see me doing it for a bit. It's not that hard really if someone wants to contribute.

**Comment by nandorojo** [comment]
Depending on the difficulty I could maybe PR this if no one else is picking it up. Ideally it would be its own prop so that you can show a custom node there with an onClick handler. Since I believe you can only render down raw HTML strings. 

How's this API?

```tsx
rowMarkerButton={{
  html: `<svg></svg>`,
  onSelect(rowIndex) {

  }
}}
```

Not exactly sure what the internals of the sheet look like, but I think that would satisfy the case for the most part. It wouldn't add support for things like the Tooltip shown in OP's post, but that would require executing some React code (or some other event handlers on the row marker element). We could implement this the way the `hasMenu` prop in column headers is implemented if the logic is easy to copy. That said, I'd need some guidance from @jassmith, as I'm not familiar with the internals.

**Comment by jassmith** [comment]
I've added all the bits required to make this possible in 6.0.0. I think the last bit now is someone to brainstorm what the api for this should be. Row markers are automatic currently... so thats a bit of an odd one.
